services/ske/wait/wait.go

```diff
@@ -148,3 +153,59 @@ func RotateCredentialsWaitHandler(ctx context.Context, a APIClientClusterInterfa
 	handler.SetTimeout(10 * time.Minute)
 	return handler
 }
+
+// StartCredentialsRotationWaitHandler will wait for credentials rotation
+func StartCredentialsRotationWaitHandler(ctx context.Context, a APIClientClusterInterface, projectId, clusterName string) *wait.AsyncActionHandler[ske.Cluster] {
+	handler := wait.New(func() (waitFinished bool, response *ske.Cluster, err error) {
+		s, err := a.GetClusterExecute(ctx, projectId, clusterName)
+		if err != nil {
+			return false, nil, err
+		}
+		state := *s.Status.CredentialsRotation.Phase
+
+		if state == CredentialsStatePrepared {
+			return true, s, nil
+		}
+
+		if state == CredentialsStatePreparing {
+			return false, nil, nil
+		}
+
+		if state == CredentialsStateNever {
+			return true, s, fmt.Errorf("starting credentials rotation failed")
+		}
```

Comment:
Why do we need a special case for the `NEVER`? According to the API documentation: `NEVER indicates that no credentials rotation has been performed using the new credentials rotation endpoints yet`, so I think we could group it into the case below where we provide the state in the error message